contracts/ERC721RExample.sol

```diff
@@ -75,26 +77,37 @@ contract ERC721RExample is ERC721A, Ownable {
         }
     }
 
-    function refund(uint256[] calldata tokenIds) external {
-        require(isRefundGuaranteeActive(), "Refund expired");
+    function refund(uint256 tokenId) external {
+        require(block.number < refundDeadlineOf(tokenId), "Refund expired");
+        require(msg.sender == ownerOf(tokenId), "Not token owner");
 
-        for (uint256 i = 0; i < tokenIds.length; i++) {
-            uint256 tokenId = tokenIds[i];
-            require(msg.sender == ownerOf(tokenId), "Not token owner");
-            require(!hasRefunded[tokenId], "Already refunded");
-            require(!isOwnerMint[tokenId], "Freely minted NFTs cannot be refunded");
-            hasRefunded[tokenId] = true;
-            transferFrom(msg.sender, refundAddress, tokenId);
-        }
+        hasRefunded[tokenId] = true;
+        _transfer(msg.sender, refundAddress, tokenId);
 
-        uint256 refundAmount = tokenIds.length * mintPrice;
+        uint256 refundAmount = refundOf(tokenId);
         Address.sendValue(payable(msg.sender), refundAmount);
     }
 
-    function getRefundGuaranteeEndTime() public view returns (uint256) {
+    function refundDeadlineOf(uint256 tokenId) public view returns (uint256) {
```

Comment:
do we want to support different refund times for different nfts in the example contract?

Reply:
We can do that if we want. In fact, we could even have an indefinite presale, and a refund period of 300000 blocks after the initial purchase.